datacube/api/query.py

```diff
@@ -79,14 +70,14 @@ def __init__(self, index=None, product=None, geopolygon=None, like=None, **searc
         Range(begin=datetime.datetime(2001, 1, 1, 0, 0, tzinfo=tzutc()), \
         end=datetime.datetime(2002, 1, 1, 23, 59, 59, 999999, tzinfo=tzutc()))
 
-        By passing in an ``index``, the search parameters will be validated as existing on the ``product``.
+        By passing in an ``index``, the search parameters will be validated as existing on the ``product``,
+        and a spatial search appropriate for the index driver can be extracted.
 
         Used by :meth:`datacube.Datacube.find_datasets` and :meth:`datacube.Datacube.load`.
 
         :param datacube.index.Index index: An optional `index` object, if checking of field names is desired.
         :param str product: name of product
-        :param geopolygon: spatial bounds of the search
-        :type geopolygon: geometry.Geometry or None
+        :type geopolygon: geometry.Geometry | None the spatial boundaries of the search
```

Comment:
This doesn't seem correct

Reply:
It's the truth but not the whole truth.

Reply:
I meant in that the `type` annotation looks like it's related to the `product` param since the `geopolygon` param was removed